### PR TITLE
SNOW-852381: [gosnowflake] Add documentation for external browser timeout

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Added the documentation for external browser authentication timeout and added the possibility to configure the parameter through DSN.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -599,6 +599,25 @@ Note: As of February 2020, Golang's official library does not support passcode-e
 For security purposes, Snowflake highly recommends that you store the passcode-encrypted private key on the disk and
 decrypt the key in your application using a library you trust.
 
+# External browser authentication
+
+The driver allows to authenticate through the external browser.
```
Comment (sfc-gh-pfus) [inline doc.go:604]: Is `through` the best word? What do you think about `using`?

File: doc.go [hunk 2]
```diff
@@ -599,6 +599,25 @@ Note: As of February 2020, Golang's official library does not support passcode-e
...
+When connection is created, the driver will open the browser window and ask the user to sign in.
+
+To enable this feature, construct the DSN with field "authenticator=EXTERNALBROWSER" or using a Config structure specifying:
+
+	config := &Config{
+		...
+		Authenticator: AuthTypeExternalBrowser,
+	}
+
+The external browser authentication implements timeout mechanism which defaults to 120s and can be changed as follows:
```
Comment (sfc-gh-pfus) [inline doc.go:615]: Maybe some words what does this timeout mean?

File: doc.go [hunk 3]
```diff
@@ -599,6 +599,25 @@ Note: As of February 2020, Golang's official library does not support passcode-e
 For security purposes, Snowflake highly recommends that you store the passcode-encrypted private key on the disk and
 decrypt the key in your application using a library you trust.
 
+# External browser authentication
+
+The driver allows to authenticate through the external browser.
+
+When connection is created, the driver will open the browser window and ask the user to sign in.
+
+To enable this feature, construct the DSN with field "authenticator=EXTERNALBROWSER" or using a Config structure specifying:
```
Comment (sfc-gh-dheyman-1) [inline doc.go:608]: perhaps `or using a Config structure as follows:` would be better?

File: doc.go [hunk 4]
```diff
@@ -599,6 +599,25 @@ Note: As of February 2020, Golang's official library does not support passcode-e
 For security purposes, Snowflake highly recommends that you store the passcode-encrypted private key on the disk and
 decrypt the key in your application using a library you trust.
 
+# External browser authentication
+
+The driver allows to authenticate through the external browser.
+
+When connection is created, the driver will open the browser window and ask the user to sign in.
```
Comment (sfc-gh-dprzybysz) [inline doc.go:606]: **a** connection

File: doc.go (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply doc.go]: Good idea, done

File: doc.go (hunk 2 above)
Comment (sfc-gh-pbulawa) [reply doc.go]: Described why the timeout is important.

File: doc.go (hunk 3 above)
Comment (sfc-gh-pbulawa) [reply doc.go]: Rephrased it as proposed.

File: doc.go (hunk 4 above)
Comment (sfc-gh-pbulawa) [reply doc.go]: Fixed the typo.

File: doc.go [hunk 5]
```diff
@@ -599,6 +599,29 @@ Note: As of February 2020, Golang's official library does not support passcode-e
...
+
+To enable this feature, construct the DSN with field "authenticator=EXTERNALBROWSER" or using a Config structure with
+following Authenticator specified:
+
+	config := &Config{
+		...
+		Authenticator: AuthTypeExternalBrowser,
+	}
+
+The external browser authentication implements timeout mechanism. This prevents the driver to hang interminably when
```
Comment (sfc-gh-pfus) [inline doc.go:616]: prevents to? Or rather prevents from hanging?

Comment (sfc-gh-pbulawa) [reply doc.go]: True, changed.

**Review by sfc-gh-pfus — APPROVED** [review]

**Review by sfc-gh-pfus — APPROVED** [review]

File: connection.go [hunk 6]
```diff
@@ -73,7 +73,7 @@ type snowflakeConn struct {
 var (
 	queryIDPattern = `[\w\-_]+`
 	queryIDRegexp  = regexp.MustCompile(queryIDPattern)
-	errMutex = &sync.Mutex{}
+	errMutex       = &sync.Mutex{}
```
Comment (sfc-gh-igarish) [inline connection.go:76]: Nit: Please check formatting in your IDE.

Comment (sfc-gh-pbulawa) [reply connection.go]: Fixed

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**